### PR TITLE
Add FindOrCreate folders

### PR DESCRIPTION
**Pull request by stefanius**

### CODE REVIEW

File: src/main/java/com/testmonitor/actions/TestCaseFolders.java [hunk 1]
```diff
@@ -200,6 +202,39 @@ public TestCaseFolder clone(TestCaseFolder source, String name) throws IOExcepti
...
+    }
+
+    /**
+     * Find a test case folder using the provided name and test folder.
+     *
+     * @param name The name of the test case folder
+     *
+     * @return Test case folders matching the provided name.
+     */
+    public ArrayList<TestCaseFolder> findByName(String name) throws IOException, URISyntaxException {
```
Comment (thijskok) [inline src/main/java/com/testmonitor/actions/TestCaseFolders.java:229]: Only root level?

This method probably needs a parent_id as well.. or overload it.

Comment (stefanius) [reply src/main/java/com/testmonitor/actions/TestCaseFolders.java:264]: Ok, I added support for trees.

File: src/main/java/com/testmonitor/actions/TestCaseFolders.java [hunk 2]
```diff
@@ -200,6 +202,87 @@ public TestCaseFolder clone(TestCaseFolder source, String name) throws IOExcepti
...
+
+        params.add(new BasicNameValuePair("project_id", this.projectId.toString()));
+        params.add(new BasicNameValuePair("parent_id", parentId.toString()));
+        params.add(new BasicNameValuePair("filter[name]", name));
+
+        return TestCaseFolderParser.parse(this.connector.get("test-case/folders", params));
+    }
+
+    /**
+     * Find a test case folder using the provided name and test folder.
```
Comment (thijskok) [inline src/main/java/com/testmonitor/actions/TestCaseFolders.java:275]: and parent folder

File: src/main/java/com/testmonitor/actions/TestCaseFolders.java [hunk 3]
```diff
@@ -200,6 +202,87 @@ public TestCaseFolder clone(TestCaseFolder source, String name) throws IOExcepti
...
+        List<NameValuePair> params = new ArrayList<>();
+
+        params.add(new BasicNameValuePair("project_id", this.projectId.toString()));
+        params.add(new BasicNameValuePair("filter[name]", name));
+
+        return TestCaseFolderParser.parse(this.connector.get("test-case/folders", params));
+    }
+
+    /**
+     * Find a test case folder using the provided name and test folder.
```
Comment (thijskok) [inline src/main/java/com/testmonitor/actions/TestCaseFolders.java:257]: and parent folder id

File: src/main/java/com/testmonitor/actions/TestCaseFolders.java [hunk 4]
```diff
@@ -200,6 +202,87 @@ public TestCaseFolder clone(TestCaseFolder source, String name) throws IOExcepti
...
+
+        if (testCaseFolders.size() > 0) {
+            return testCaseFolders.get(0);
+        }
+
+        return this.create(name);
+    }
+
+    /**
+     * Find a test case folder using the provided name or create a new one.
```
Comment (thijskok) [inline src/main/java/com/testmonitor/actions/TestCaseFolders.java:223]: the provided name and parent folder

File: src/main/java/com/testmonitor/actions/TestCaseFolders.java [hunk 5]
```diff
@@ -123,14 +125,14 @@ public TestCaseFolder create(String name) throws IOException {
     }
```
Comment (thijskok) [inline src/main/java/com/testmonitor/actions/TestCaseFolders.java:125]: ⬆️ The test case folder name you want to create => The name of the test case folder

**Review by thijskok — APPROVED** [review]

**Pull request merged**